File: Snooper/src/main/java/com/prateekj/snooper/formatter/JsonResponseFormatter.java
package com.prateekj.snooper.formatter;

import com.prateekj.snooper.utils.Logger;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;
import org.json.JSONTokener;

public class JsonResponseFormatter implements ResponseFormatter {

  private static final String TAG = JsonResponseFormatter.class.getSimpleName();
  private static final int INDENT_SPACES = 4;

  @Override
  public String format(String response) {
    try {
      Object json = new JSONTokener(response).nextValue();
      if (json instanceof JSONObject) {
        return new JSONObject(response).toString(INDENT_SPACES);
      }
      return new JSONArray(response).toString(INDENT_SPACES);
    } catch (JSONException e) {
      e.printStackTrace();
      Logger.e(TAG, e.getMessage(), e);
    }
    return response;
  }
}

File: Snooper/src/main/java/com/prateekj/snooper/formatter/JsonResponseFormatter.kt
package com.prateekj.snooper.formatter

import com.prateekj.snooper.utils.Logger

import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject
import org.json.JSONTokener

class JsonResponseFormatter : ResponseFormatter {

  override fun format(response: String): String {
    try {
      val json = JSONTokener(response).nextValue()
      return if (json is JSONObject) {
        JSONObject(response).toString(INDENT_SPACES)
      } else JSONArray(response).toString(INDENT_SPACES)
    } catch (e: JSONException) {
      e.printStackTrace()
      Logger.e(TAG, e.message, e)
    }

    return response
  }

  companion object {
    private val TAG = JsonResponseFormatter::class.java.simpleName
    private const val INDENT_SPACES = 4
  }
}
